src/main/java/org/zalando/nakadi/controller/HealthCheckController.java

```diff
@@ -12,8 +14,18 @@
 @RequestMapping(value = "/health", produces = TEXT_PLAIN_VALUE)
 public class HealthCheckController {
 
+    private final TerminationService terminationService;
+
+    @Autowired
+    public HealthCheckController(final TerminationService terminationService) {
+        this.terminationService = terminationService;
+    }
+
     @RequestMapping(method = GET)
     public ResponseEntity<String> healthCheck() {
+        if (terminationService.isTerminating()) {
+            return ResponseEntity.status(HttpStatus.IM_A_TEAPOT_418).build();
```

Comment:
Is this meant to be used by Nakadi Proxy? When maintaining the list of healthy Nakadi instances, those who report 418 during health check will not be added to the final result but at the same time not marked as unhealthy?